pkg/test/framework/components/istio/operator.go

```diff
@@ -773,6 +760,47 @@ func (i *operatorComponent) configureDirectAPIServiceAccessBetweenClusters(ctx r
 	return nil
 }
 
+func getTargetClusterListForCluster(targetClusters []cluster.Cluster, c cluster.Cluster) []cluster.Cluster {
+	var outClusters []cluster.Cluster
+	scopes.Framework.Infof("Secret from cluster: %s will be placed in following clusters", c.Name())
+	for _, cc := range targetClusters {
+		if c.IsConfig() && !cc.IsConfig() && !cc.IsRemote() {
```

Comment:
Simpler check and also should handle case of more than one external control plane in the topology.

```suggestion
		if c.IsConfig() && cc.IsExternalControlPlane() && c.Primary().Name() == cc.Name() {
```

Reply:
@frankbu ok. will change to the suggestion.

Reply:
@frankbu Actually the suggestion and the original are not the same. The `else if` check was to make sure it is not adding secret to itself. but the original can be simplified. The point is that c.Primary().Name() does not always equal to c.Name().

Reply:
@frankbu please see the new commit. Thanks.

Reply:
> The point is that c.Primary().Name() does not always equal to c.Name().

My suggestion, above, is comparing to `cc.Name()`, not `c.Name()`.